### ISSUE TITLE
[Bug] must press x button at the end of video

### ISSUE THREAD

**Issue by hotdog5785** [qnblackcat/uYouPlus#54]
**Device info:**
- iOS/iPadOS version: 15.3
- Device model: iphone 13 pro
- Sideload tool: altsotre
- Version of uYouPlus:  16.46.5

**Describe the bug:**
at the end of the video, i must press x to end fullscreen video and watch other videos
![1DE21EA2-6515-4E01-A707-A3B033FDF2F4](https://user-images.githubusercontent.com/99060747/152626939-7442adad-825f-408f-9e53-a4e3d5e8cc48.png)

without press x button, i can't do any additional actions in app

when i press x button, then i can end fullscreen video and navigate other vidoes.

![AEADA9BC-B721-48FD-B0FB-EADAB7E90FA3](https://user-images.githubusercontent.com/99060747/152626982-644c6777-bed4-47cd-8b6e-14d574ed3a08.png)

i think this bug is related to autoplay option at the end of the video in youtube app.
I turn off video autoplay option in youtube original settings.

**Comment by qnblackcat** [comment]
Do you have:
1. **Autoplay** enabled?
2. **Related Videos at the end the videos** disabled in uYou’s settings > Video Controls Overlay?

**Comment by hotdog5785** [comment]
1. disabled autoplay in yt original settings
2. disabled related videos at the end the videos option in uyou settings

after some experiment with your comment, i think uyou option for 'related videos at the end the videos' is the reason of this situation. uyou succesfully erased related videos, but leave behind its background.

**Comment by hotdog5785** [comment]
@qnblackcat

disabled autoplay in yt original settings
disabled related videos at the end the videos option in uyou settings
after some experiment with your comment, i think uyou option for 'related videos at the end the videos' is the reason of this situation. uyou succesfully erased related videos, but leave behind its transparent background.

**Comment by qnblackcat** [comment]
@hotdog5785 please try [this version](https://27man-my.sharepoint.com/:u:/g/personal/qn____27man_onmicrosoft_com/EQ9yJVZBrX1Jq5atwsdSOdwBpsQqF8NCvCqfNYy_PoHsmQ?download=1)

**Comment by qnblackcat** [comment]
```
%hook YTMainAppVideoPlayerOverlayViewController
- (void)updateRelatedVideos {}
%end
```
This is the workaround

**Comment by abheizthe** [comment]
this is happened again in v17.23.6-2.1
autoplay disable
related video disable

**Comment by qnblackcat** [comment]
> this is happened again in v17.23.6-2.1
> autoplay disable
> related video disable

I removed the fix because it causes issue for people who never turn on/off “Hide related video…” in uYou. 
Anyway, I mentioned the fix above. You can build your own IPA with the new code.

**Comment by qnblackcat** [comment]
Should be fixed in v17.24.4-2.1 and higher. Please re-open the thread if the issue persists
